### DISCUSSION TITLE
How to subdivise the codemagic.yaml file into multiple yaml files?

### DISCUSSION THREAD

**Discussion by mregnauld** [codemagic-ci-cd/codemagic-docs#1331] [Q&A] [answered]
My `codemagic.yaml` file has many workflows and is getting pretty big and hard to read.

Is there a way to subdivise that file in multiple files (one for each workflow for example), and include them in the `codemagic.yaml` file? It's something possible with AWS for example, and it's very convenient.

Thanks.

**Comment by himesh-cm** [comment, marked as answer]
Hi @mregnauld, currently, it is not possible to create separate `codemagic.yaml` files. However, you can make the file more readable by using anchors and aliases. 

I'll share this feedback with the team.

Thanks for using Codemagic.

**Reply by mregnauld** [reply]
Yes, I use the `definitions` a lot, and sure it helps.
To be honest, I'm not sure if it's a *must have* feature, but definitely a very *nice to have* :)

**Reply by himesh-cm** [reply]
Hi @mregnauld , one of our devs suggested the below approach.
You can extract scripts to separate files and store them say `./ci-script`
In your `codemagic.yaml` file you can create a different workflow for each and run them as per your requirements.

e.g.
```
- name: Make GitHub release
  script: ./ci-scripts/github-release.sh
```

Let me know if this helps.

**Reply by mregnauld** [reply]
Oh ok, I didn't think about that one, thanks!
